libraries/tools/kotlin-stdlib-gen/src/templates/dsl/CommonTypes.kt

```diff
@@ -106,20 +106,22 @@ enum class Platform {
     Common,
     JVM,
     JS,
-    Native
+    Native,
+    Wasm,
```

Comment:
It seems that `Wasm` shares a lot of generated implementations with `Native`. Does it make sense then to make it a subcategory of `Native` platform?

Reply:
I don't have an opinion on this.

Currently Wasm has a lot of `TODO("...")` which is different from Native. 
For common parts I taught `on` to take a vararg, so you can conveniently, yet explicitly, write `on(Platform.Native, Platform.Wasm)`

Reply:
See my attempt on reusing the Native generated functions for Wasm in PR #3883 to this branch. 
Later I plan to make the `Platform`  enum a sealed hierarchy to avoid this unnatural wasm backend for native target.